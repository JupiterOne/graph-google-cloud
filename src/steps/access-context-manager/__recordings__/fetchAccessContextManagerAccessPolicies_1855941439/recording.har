{
  "log": {
    "_recordingName": "fetchAccessContextManagerAccessPolicies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 572,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 572,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:23 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.123Z",
        "time": 212,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 212
        }
      },
      {
        "_id": "e7ae7a9d3285d95d4330a7c239aa61b7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.15.4 auth/7.1.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "accesscontextmanager.googleapis.com"
            }
          ],
          "headersSize": 581,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "parent",
              "value": "organizations/958457776463"
            }
          ],
          "url": "https://accesscontextmanager.googleapis.com/v1/accessPolicies?parent=organizations%2F958457776463"
        },
        "response": {
          "bodySize": 369,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 369,
            "text": "{\"accessPolicies\":[{\"name\":\"accessPolicies/368605946207\",\"parent\":\"organizations/958457776463\",\"title\":\"default policy\",\"etag\":\"64186ca0c82b01b4\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:24 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.374Z",
        "time": 829,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 829
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
