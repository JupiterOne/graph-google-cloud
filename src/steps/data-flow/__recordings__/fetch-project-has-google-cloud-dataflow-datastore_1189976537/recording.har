{
  "log": {
    "_recordingName": "fetch-project-has-google-cloud-dataflow-datastore",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 727,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "727"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1231,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1231,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 07 May 2024 12:39:34 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-07T12:39:34.765Z",
        "time": 234,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 234
        }
      },
      {
        "_id": "23a717153d5a0e7db6eb50c807895f67",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "cloudresourcemanager.googleapis.com"
            }
          ],
          "headersSize": 1322,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://cloudresourcemanager.googleapis.com/v3/projects/j1-gc-integration-dev-v3"
        },
        "response": {
          "bodySize": 336,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 336,
            "text": "{\"name\":\"projects/167984947943\",\"parent\":\"folders/306329820365\",\"projectId\":\"j1-gc-integration-dev-v3\",\"state\":\"ACTIVE\",\"displayName\":\"j1-gc-integration-dev-v3\",\"createTime\":\"2021-05-31T15:15:56.125Z\",\"updateTime\":\"2024-02-27T16:17:58.322524Z\",\"etag\":\"W/\\\"8f56735f6f3d890d\\\"\",\"labels\":{\"firebase\":\"enabled\"}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 07 May 2024 12:39:36 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 367,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-07T12:39:35.016Z",
        "time": 1720,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1720
        }
      },
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 727,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/9.4.1"
            },
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "727"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 293,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 1164,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 1164,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 07 May 2024 12:39:36 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 390,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-07T12:39:36.782Z",
        "time": 185,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 185
        }
      },
      {
        "_id": "45fdb1f844a17b6d6ec371710879c815",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/7.0.1 gl-node/18.17.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/7.0.1 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "name": "host",
              "value": "dataflow.googleapis.com"
            }
          ],
          "headersSize": 1305,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dataflow.googleapis.com/v1b3/projects/j1-gc-integration-dev-v3/jobs"
        },
        "response": {
          "bodySize": 739,
          "content": {
            "encoding": "utf-8",
            "mimeType": "application/json; charset=UTF-8",
            "size": 739,
            "text": "{\"jobs\":[{\"id\":\"2024-05-02_05_25_02-17305733872805367843\",\"projectId\":\"j1-gc-integration-dev-v3\",\"name\":\"test-job3\",\"type\":\"JOB_TYPE_BATCH\",\"currentState\":\"JOB_STATE_FAILED\",\"currentStateTime\":\"2024-05-02T13:28:54.170517Z\",\"createTime\":\"2024-05-02T12:25:06.098727Z\",\"location\":\"us-central1\",\"jobMetadata\":{\"sdkVersion\":{\"version\":\"2.55.1\",\"versionDisplayName\":\"Apache Beam SDK for Java\",\"sdkSupportStatus\":\"SUPPORTED\"}},\"startTime\":\"2024-05-02T12:25:06.098727Z\"},{\"id\":\"2024-04-29_06_54_21-11420805499054724452\",\"projectId\":\"j1-gc-integration-dev-v3\",\"name\":\"test-metron-job2\",\"currentState\":\"JOB_STATE_FAILED\",\"currentStateTime\":\"2024-04-29T14:01:46.604199Z\",\"createTime\":\"2024-04-29T13:54:22.225266Z\",\"location\":\"us-central1\",\"startTime\":\"2024-04-29T13:54:22.225266Z\"},{\"id\":\"2024-04-29_06_40_47-701576625030004708\",\"projectId\":\"j1-gc-integration-dev-v3\",\"name\":\"test-metron-job1\",\"currentState\":\"JOB_STATE_FAILED\",\"currentStateTime\":\"2024-04-29T13:46:31.741443Z\",\"createTime\":\"2024-04-29T13:40:48.311738Z\",\"location\":\"us-central1\",\"startTime\":\"2024-04-29T13:40:48.311738Z\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 07 May 2024 12:39:38 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 367,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-05-07T12:39:36.977Z",
        "time": 1837,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1837
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
