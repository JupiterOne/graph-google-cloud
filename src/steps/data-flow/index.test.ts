import {
    createMockStepExecutionContext,
    Recording,
  } from '@jupiterone/integration-sdk-testing';
  import {
    fetchGoogleCloudDataFlowDataStore,
    fetchGoogleCloudDataFlowJob,
    fetchGoogleCloudDataFlowService,
    fetchGoogleCloudDataFlowSnapshot,
  } from '.';
  import { integrationConfig } from '../../../test/config';
  import { setupGoogleCloudRecording } from '../../../test/recording';
  import { IntegrationConfig } from '../../types';
  import {
    GOOGLE_CLOUD_DATAFLOW_DATASTORE_TYPE,
    GOOGLE_CLOUD_DATAFLOW_JOB_CLASS,
    GOOGLE_CLOUD_DATAFLOW_SNAPSHOT_TYPE,
    GOOGLE_CLOUD_DATAFLOW_TYPE,
  } from './constants';
  
  const tempNewAccountConfig = {
    ...integrationConfig,
    serviceAccountKeyFile: integrationConfig.serviceAccountKeyFile.replace(
      'j1-gc-integration-dev-v2',
      'j1-gc-integration-dev-v3',
    ),
    serviceAccountKeyConfig: {
      ...integrationConfig.serviceAccountKeyConfig,
      project_id: 'j1-gc-integration-dev-v3',
    },
  };
  
  describe('#fetchGoogleCloudDataFlowService', () => {
    let recording: Recording;
  
    beforeEach(() => {
      recording = setupGoogleCloudRecording({
        directory: __dirname,
        name: 'fetchGoogleCloudDataFlowService',
      });
    });
  
    afterEach(async () => {
      await recording.stop();
    });
  
    test('should collect data', async () => {
      const context = createMockStepExecutionContext<IntegrationConfig>({
        instanceConfig: tempNewAccountConfig,
      });
  
      await fetchGoogleCloudDataFlowService(context);
  
      expect({
        numCollectedEntities: context.jobState.collectedEntities.length,
        numCollectedRelationships: context.jobState.collectedRelationships.length,
        collectedEntities: context.jobState.collectedEntities,
        collectedRelationships: context.jobState.collectedRelationships,
        encounteredTypes: context.jobState.encounteredTypes,
      }).toMatchSnapshot();
  
      expect(
        context.jobState.collectedEntities.filter(
          (e) => e.type === GOOGLE_CLOUD_DATAFLOW_TYPE,
        ),
      ).toMatchGraphObjectSchema({
        _class: ['Service'],
        schema: {
          additionalProperties: false,
          properties: {
            _type: { const: 'google_cloud_dataflow' },
            name: { type: 'string' },
            category: { type: 'array' },
            function: { type: 'array' },
          },
        },
      });
    });
  });
  
  describe('#fetchGoogleCloudDataFlowJob', () => {
    let recording: Recording;
  
    beforeEach(() => {
      recording = setupGoogleCloudRecording({
        directory: __dirname,
        name: 'fetchGoogleCloudDataFlowJob',
      });
    });
  
    afterEach(async () => {
      await recording.stop();
    });
  
    test('should collect data', async () => {
      const context = createMockStepExecutionContext<IntegrationConfig>({
        instanceConfig: tempNewAccountConfig,
      });
  
      await fetchGoogleCloudDataFlowJob(context);
  
      expect({
        numCollectedEntities: context.jobState.collectedEntities.length,
        numCollectedRelationships: context.jobState.collectedRelationships.length,
        collectedEntities: context.jobState.collectedEntities,
        collectedRelationships: context.jobState.collectedRelationships,
        encounteredTypes: context.jobState.encounteredTypes,
      }).toMatchSnapshot();
  
      expect(
        context.jobState.collectedEntities.filter(
          (e) => e.type === GOOGLE_CLOUD_DATAFLOW_JOB_CLASS,
        ),
      ).toMatchGraphObjectSchema({
        _class: ['Workflow'],
        schema: {
          additionalProperties: false,
          properties: {
            _type: { const: 'google_cloud_dataflow_job' },
            name: { type: 'string' },
            displayname: { type: 'string' },
            description: { type: 'string' },
          },
        },
      });
    });
  });
  
  describe('#fetchGoogleCloudDataFlowDataStore', () => {
    let recording: Recording;
  
    beforeEach(() => {
      recording = setupGoogleCloudRecording({
        directory: __dirname,
        name: 'fetchGoogleCloudDataFlowDataStore',
      });
    });
  
    afterEach(async () => {
      await recording.stop();
    });
  
    test('should collect data', async () => {
      const context = createMockStepExecutionContext<IntegrationConfig>({
        instanceConfig: tempNewAccountConfig,
      });
  
      await fetchGoogleCloudDataFlowDataStore(context);
  
      expect({
        numCollectedEntities: context.jobState.collectedEntities.length,
        numCollectedRelationships: context.jobState.collectedRelationships.length,
        collectedEntities: context.jobState.collectedEntities,
        collectedRelationships: context.jobState.collectedRelationships,
        encounteredTypes: context.jobState.encounteredTypes,
      }).toMatchSnapshot();
  
      expect(
        context.jobState.collectedEntities.filter(
          (e) => e.type === GOOGLE_CLOUD_DATAFLOW_DATASTORE_TYPE,
        ),
      ).toMatchGraphObjectSchema({
        _class: ['Datastore'],
        schema: {
          additionalProperties: false,
          properties: {
            _type: { const: 'google_cloud_dataflow_datastore' },
            name: { type: 'string' },
            encrypted: { type: 'string' },
            projectId: { type: 'string' },
          },
        },
      });
    });
  });
  
  describe('#fetchGoogleCloudDataFlowSnapshot', () => {
    let recording: Recording;
  
    beforeEach(() => {
      recording = setupGoogleCloudRecording({
        directory: __dirname,
        name: 'fetchGoogleCloudDataFlowSnapshot',
      });
    });
  
    afterEach(async () => {
      await recording.stop();
    });
  
    test('should collect data', async () => {
      const context = createMockStepExecutionContext<IntegrationConfig>({
        instanceConfig: tempNewAccountConfig,
      });
  
      await fetchGoogleCloudDataFlowSnapshot(context);
      expect({
        numCollectedEntities: context.jobState.collectedEntities.length,
        numCollectedRelationships: context.jobState.collectedRelationships.length,
        collectedEntities: context.jobState.collectedEntities,
        collectedRelationships: context.jobState.collectedRelationships,
        encounteredTypes: context.jobState.encounteredTypes,
      }).toMatchSnapshot();
  
      expect(
        context.jobState.collectedEntities.filter(
          (e) => e.type === GOOGLE_CLOUD_DATAFLOW_SNAPSHOT_TYPE,
        ),
      ).toMatchGraphObjectSchema({
        _class: ['Database'],
        schema: {
          additionalProperties: false,
          properties: {
            _type: { const: 'google_bigtable_backup' },
            name: { type: 'string' },
            projectId: { type: 'string' },
            classification: { type: 'string' },
            encrypted: { type: 'boolean' },
            jobId: { type: 'string' },
          },
        },
      });
    });
  });
  
  