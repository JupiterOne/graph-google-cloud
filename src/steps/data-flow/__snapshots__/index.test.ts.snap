// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchGoogleCloudDataFlowDataStore should collect data 1`] = `
{
  "collectedEntities": [],
  "collectedRelationships": [],
  "encounteredTypes": [],
  "numCollectedEntities": 0,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchGoogleCloudDataFlowJob should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Workflow",
      ],
      "_key": "2024-05-02_05_25_02-17305733872805367843",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "createTime": "2024-05-02T12:25:06.098727Z",
            "currentState": "JOB_STATE_FAILED",
            "currentStateTime": "2024-05-02T13:28:54.170517Z",
            "id": "2024-05-02_05_25_02-17305733872805367843",
            "jobMetadata": {
              "sdkVersion": {
                "sdkSupportStatus": "SUPPORTED",
                "version": "2.55.1",
                "versionDisplayName": "Apache Beam SDK for Java",
              },
            },
            "location": "us-central1",
            "name": "test-job3",
            "projectId": "j1-gc-integration-dev-v3",
            "startTime": "2024-05-02T12:25:06.098727Z",
            "type": "JOB_TYPE_BATCH",
          },
        },
      ],
      "_type": "google_cloud_dataflow_job",
      "createdOn": undefined,
      "description": undefined,
      "displayName": "test-job3",
      "id": "2024-05-02_05_25_02-17305733872805367843",
      "name": "test-job3",
    },
    {
      "_class": [
        "Workflow",
      ],
      "_key": "2024-04-29_06_54_21-11420805499054724452",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "createTime": "2024-04-29T13:54:22.225266Z",
            "currentState": "JOB_STATE_FAILED",
            "currentStateTime": "2024-04-29T14:01:46.604199Z",
            "id": "2024-04-29_06_54_21-11420805499054724452",
            "location": "us-central1",
            "name": "test-metron-job2",
            "projectId": "j1-gc-integration-dev-v3",
            "startTime": "2024-04-29T13:54:22.225266Z",
          },
        },
      ],
      "_type": "google_cloud_dataflow_job",
      "createdOn": undefined,
      "description": undefined,
      "displayName": "test-metron-job2",
      "id": "2024-04-29_06_54_21-11420805499054724452",
      "name": "test-metron-job2",
    },
    {
      "_class": [
        "Workflow",
      ],
      "_key": "2024-04-29_06_40_47-701576625030004708",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "createTime": "2024-04-29T13:40:48.311738Z",
            "currentState": "JOB_STATE_FAILED",
            "currentStateTime": "2024-04-29T13:46:31.741443Z",
            "id": "2024-04-29_06_40_47-701576625030004708",
            "location": "us-central1",
            "name": "test-metron-job1",
            "projectId": "j1-gc-integration-dev-v3",
            "startTime": "2024-04-29T13:40:48.311738Z",
          },
        },
      ],
      "_type": "google_cloud_dataflow_job",
      "createdOn": undefined,
      "description": undefined,
      "displayName": "test-metron-job1",
      "id": "2024-04-29_06_40_47-701576625030004708",
      "name": "test-metron-job1",
    },
  ],
  "collectedRelationships": [],
  "encounteredTypes": [
    "google_cloud_dataflow_job",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchGoogleCloudDataFlowService should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Service",
      ],
      "_key": "google_cloud_dataflow:j1-gc-integration-dev-v3",
      "_rawData": [],
      "_type": "google_cloud_dataflow",
      "category": [
        "network",
        "security",
      ],
      "createdOn": undefined,
      "displayName": "Google Cloud Data Flow",
      "function": [
        "workflow",
        "networking",
      ],
      "name": "Google Cloud Data Flow",
    },
  ],
  "collectedRelationships": [],
  "encounteredTypes": [
    "google_cloud_dataflow",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;

exports[`#fetchGoogleCloudDataFlowSnapshot should collect data 1`] = `
{
  "collectedEntities": [],
  "collectedRelationships": [],
  "encounteredTypes": [],
  "numCollectedEntities": 0,
  "numCollectedRelationships": 0,
}
`;
